### DISCUSSION TITLE
[bug] - Storefront API "query" variable simply not working properly

### DISCUSSION THREAD

**Discussion by joelpierre** [Shopify/storefront-api-feedback#195] [Help]
I have posted this on shopify forum here [[forum]](https://community.shopify.com/c/hydrogen-headless-and-storefront/storefront-quot-query-quot-field-not-returning-results-for-any/td-p/2127411) However as it is API related I assume I will get a quicker and better response here. 

We have a number of services that rely on the storefront API and utilise the "query" variable, all of which have been having significant issues since 30th June.

A simple product query using 
```json
"variables":{"first":20,"sortKey":"ID","query":"variants:['sku:dress'] OR title:dress","reverse":true}
```

Returns no results
<img width="494" alt="Screenshot 2023-07-04 at 12 18 39" src="https://github.com/Shopify/storefront-api-feedback/assets/8969851/806968b6-5c93-483e-a1c0-9f23ad9a7537">

However the same query in the shopify admin search returns products
<img width="657" alt="Screenshot 2023-07-04 at 12 17 47" src="https://github.com/Shopify/storefront-api-feedback/assets/8969851/9ac76ce0-26b4-4afa-b455-3d9c56922544">

We had other queries which leveraged the "query" var but we have omitted it as a bandaid in order to get "something" showing. 

The API token has all available scopes and we have swapped out many tokens to strangely get:
1. different product results
2. the same no items when using the query variable
 
We have also noticed SIGNIFICANT delays in Shopify's search indexes over the last couple of weeks/month, in which products simply do not show for hours if not days. 

How do we get this matter resolved ASAP.

**Comment by johncraigcole** [comment]
@joelpierre can you share a ticket # where this was reported to support? Also a recent `x-request-id` from the response headers of a request that didn't return the expected result would also be helpful.

**Comment by joelpierre** [comment]
Hi @johncraigcole  the ticket number for the escalated request is 39675493.

Other tickets are: 2725631, 2575598, 2701683, 2701529. 

Request ID for a search is: `bf2504fc-bb06-48d8-976a-67febd85f154`

Its a search for "bamboo cookies"

In Shopify UI:
<img width="609" alt="Screenshot 2023-07-05 at 01 51 02" src="https://github.com/Shopify/storefront-api-feedback/assets/8969851/de4dae55-208d-423a-aac1-1d0b1459b05d">

via the API
<img width="1459" alt="Screenshot 2023-07-05 at 01 51 59" src="https://github.com/Shopify/storefront-api-feedback/assets/8969851/f27af462-269f-4a15-9454-fc1b6c39d1b9">

From debugging for quite some time I noticed I have this issue largely on products with no variants. It seems previously these variants used to have the title of the product that wrapped it but now the only title that yields results for single variant products is "default title" 🙃 

And this is only since the 30th June it would seem as reported by our content team.

**Comment by johncraigcole** [comment]
Thanks @joelpierre I'll do some digging and get back to you.

**Comment by nn4u** [comment]
Hello, I have also some troubles when I make a query by tag.

I have products with long tag like "col_rel:this-is-the-handle-of-another-product" or more common like look:1000.

`products(first: 10, query: "tag:look:1000")` will work but not `products(first: 10, query: "tag:col_rel:this-is-the-handle-of-another-product")`

**Reply by johncraigcole** [reply]
@nn4u can you share the `x-request-id` response header from a Storefront API request that is not returning the results you expect.

**Reply by nn4u** [reply]
Thanks, this is the request id 9f373940-38c4-47d0-a4fe-537b49a8e1a0

**Reply by johncraigcole** [reply]
Thanks @nn4u I would try putting the tag value in single quotes. 

```
products(first: 10, query: "tag:'col_rel:this-is-the-handle-of-another-product'"){...}
```
If that doesn't work you'll have to contact support as there's nothing from the logs indicating a problem on our end. In my own testing with the tag value `col_rel:Camiseta-historie-amour-Color-Crudo-XENT930333` surrounding that with single quotes in the query worked for me.

**Reply by nn4u** [reply]
Thank you !

**Comment by guillaumelachaud** [comment]
I have created 3 automatic collections on my store. When trying to fetch these collections with the following query, I get an empty array:

```
curl -X POST \
  https://myshop.myshopify.com/api/2023-07/graphql.json \
  -H 'Content-Type: application/graphql' \
  -H 'X-Shopify-Storefront-Access-Token: TOKEN' \
  -d '
  query FeaturedCollections {
    collections(first: 3, query: "collection_type:smart") {
      nodes {
        id
        title
        handle
        image {
          altText
          width
          height
          url
        }
      }
    }
  }
  '
```

Two things to note:
 - If I remove the query variable, I do receive my 3 collections properly
 - The same query in the Shopify GraphiQL App works without problems, which is really odd. I was expecting the app to hit the same API but it seems it's not the case.


Here is a x-request-id of a response without the proper results: `63374269-34f5-4971-9575-5e4713d9248f`

**Reply by johncraigcole** [reply]
@guillaumelachaud have you double checked to make sure that the app being used for the request-id above has collections published to it? This can be verified by going to a given collection, and on the right clicking `Manage` this will load a dialog which should allow you to see if the collection is published to your app.

**Reply by guillaumelachaud** [reply]
Yes, looks published on all channels:
<img width="308" alt="CleanShot 2023-07-05 at 20 47 20@2x" src="https://github.com/Shopify/storefront-api-feedback/assets/2076017/b11faacb-8d42-43bf-9991-2f7d23c8becb">

**Reply by johncraigcole** [reply]
The request-id provided some context but to validate this is the same issue relating to the original requester, I'll need you to contact support if you haven't already so you're authenticated on the account. From there if you can share the ticket number I'll attach it to the internal issue.

**Comment by johncraigcole** [comment]
Hey @guillaumelachaud @joelpierre we shipped a fix a short time ago which I believe should resolve your issues. Please let me know if you're not seeing the problems as resolved.

**Reply by guillaumelachaud** [reply]
I confirm the fix to work for me, thanks!

**Reply by joelpierre** [reply]
My team has confirmed the fix 🙌🏽. 

Thanks will monitor to see if I see any regressions 👍🏽

**Comment by joelpierre** [comment]
@johncraigcole  - I have been told there are still intermittent issues with this, and also looks like a general cache layer issue across fetches. I saw in another thread that there were issues with your cache layers in Europe. Could this still be impacting us?

**Reply by johncraigcole** [reply]
I'm not aware of any continuing issues. All the reports that came up around the time of your issues, I've followed up 
with and have gotten confirmation all is good across the board. If you have examples, please send the x-request-id's.
